pytorch_lightning/tuner/batch_size_scaling.py

```diff
@@ -85,59 +86,44 @@ def scale_batch_size(
     log.info(f"Finished batch size finder, will continue with full run using batch size {new_size}")
 
     # Restore initial state of model
-    if trainer.is_global_zero:
-        trainer.checkpoint_connector.restore(str(save_path))
-        fs = get_filesystem(str(save_path))
-        if fs.exists(save_path):
-            fs.rm(save_path)
-
-    # Finish by resetting variables so trainer is ready to fit model
-    __scale_batch_restore_params(trainer)
+    trainer.checkpoint_connector.restore(ckpt_path)
+    trainer.strategy.remove_checkpoint(ckpt_path)
+    __scale_batch_restore_params(trainer, params)
+
     if trainer.progress_bar_callback:
         trainer.progress_bar_callback.enable()
 
     return new_size
 
 
-def __scale_batch_dump_params(trainer: "pl.Trainer") -> None:
-    # Prevent going into infinite loop
-    trainer.__dumped_params = {
-        "auto_lr_find": trainer.auto_lr_find,
-        "current_epoch": trainer.current_epoch,
-        "global_step": trainer.global_step,
-        "max_steps": trainer.max_steps,
+def __scale_batch_dump_params(trainer: "pl.Trainer") -> Dict[str, Any]:
+    return {
+        "max_steps": trainer.fit_loop.max_steps,
         "logger": trainer.logger,
         "callbacks": trainer.callbacks,
-        "checkpoint_callback": trainer.checkpoint_callback,
         "auto_scale_batch_size": trainer.auto_scale_batch_size,
+        "auto_lr_find": trainer.auto_lr_find,
         "limit_train_batches": trainer.limit_train_batches,
-        "model": trainer.model,
```

Comment:
might need to update a test here: https://github.com/PyTorchLightning/pytorch-lightning/blob/f7f835fa0e22dc34631aa43d54bf73304c9bda22/tests/tuner/test_scale_batch_size.py#L98

same for lr_finder.

Reply:
What do you mean? The test is working, no?

Reply:
yes.. I mean since a few of the arguments are removed here, maybe need to remove them from the test too?

Reply:
This test was helpful to validate my implementation (had some bugs before) so I prefer keeping those